cpp/non_iid_main.cpp

```diff
@@ -217,105 +217,101 @@ int main(int argc, char* argv[]) {
+        tc632.ret_min_entropy = ret_min_entropy;
     }
 
     if (initial_entropy && (data.alph_size == 2)) {
         ret_min_entropy = collision_test(data.symbols, data.len, verbose, "Literal");
         if (verbose == 1) printf("\tCollision Test Estimate = %f / 1 bit(s)\n", ret_min_entropy);
         H_original = min(ret_min_entropy, H_original);
+        tc632.ret_min_entropy = ret_min_entropy;
     }
 
-    NonIidTestCase tc632;
-    tc632.h_bitstring = H_bitstring;
-    tc632.h_original = H_original;
-    tc632.ret_min_entropy = ret_min_entropy;
-    tc632.data_word_size = data.word_size;
-    tc632.lrs_res = lrs_res;
     tc632.testCaseNumber = "Estimate entropy with Collision Test (for bit strings only)";
     testRun.testCases.push_back(tc632);
 
     // Section 6.3.3 - Estimate entropy with Markov Test (for bit strings only)
+    NonIidTestCase tc633;
+
     if (((data.alph_size > 2) || !initial_entropy)) {
         ret_min_entropy = markov_test(data.bsymbols, data.blen, verbose, "Bitstring");
         if (verbose == 1) printf("\tMarkov Test Estimate (bit string) = %f / 1 bit(s)\n", ret_min_entropy);
         H_bitstring = min(ret_min_entropy, H_bitstring);
+        tc633.ret_min_entropy = ret_min_entropy;
     }
 
     if (initial_entropy && (data.alph_size == 2)) {
         ret_min_entropy = markov_test(data.symbols, data.len, verbose, "Literal");
         if (verbose == 1) printf("\tMarkov Test Estimate = %f / 1 bit(s)\n", ret_min_entropy);
         H_original = min(ret_min_entropy, H_original);
+        tc633.ret_min_entropy = ret_min_entropy;
     }
 
-    NonIidTestCase tc633;
-    tc633.h_bitstring = H_bitstring;
-    tc633.h_original = H_original;
-    tc633.ret_min_entropy = ret_min_entropy;
-    tc633.data_word_size = data.word_size;
-    tc633.lrs_res = lrs_res;
     tc633.testCaseNumber = "Estimate entropy with Markov Test (for bit strings only)";
     testRun.testCases.push_back(tc633);
 
     // Section 6.3.4 - Estimate entropy with Compression Test (for bit strings only)
+    NonIidTestCase tc634;
+
     if (((data.alph_size > 2) || !initial_entropy)) {
         ret_min_entropy = compression_test(data.bsymbols, data.blen, verbose, "Bitstring");
         if (ret_min_entropy >= 0) {
             if (verbose == 1) printf("\tCompression Test Estimate (bit string) = %f / 1 bit(s)\n", ret_min_entropy);
             H_bitstring = min(ret_min_entropy, H_bitstring);
+            tc634.ret_min_entropy = ret_min_entropy;
         }
     }
 
     if (initial_entropy && (data.alph_size == 2)) {
         ret_min_entropy = compression_test(data.symbols, data.len, verbose, "Literal");
         if (ret_min_entropy >= 0) {
             if (verbose == 1) printf("\tCompression Test Estimate = %f / 1 bit(s)\n", ret_min_entropy);
-            H_original = min(ret_min_entropy, H_original);    
+            H_original = min(ret_min_entropy, H_original);
+            tc634.ret_min_entropy = ret_min_entropy;
```

Comment:
Similarly, if both literal and bitstring assessments are conducted, only the literal assessment is saved.